src/network/tcp.py

```diff
@@ -67,18 +68,25 @@ def __init__(self, address=None, sock=None):
             self.connect(self.destination)
             logger.debug("Connecting to %s:%i", self.destination.host, self.destination.port)
         encodedAddr = protocol.encodeHost(self.destination.host)
-        if protocol.checkIPAddress(encodedAddr, True) and not protocol.checkSocksIP(self.destination.host):
-            self.local = True
-        else:
-            self.local = False
-        #shared.connectedHostsList[self.destination] = 0
-        ObjectTracker.__init__(self)
+        self.local = all([
+            protocol.checkIPAddress(encodedAddr, True),
+            not protocol.checkSocksIP(self.destination.host)
+        ])
+        ObjectTracker.__init__(self)  # pylint: disable=non-parent-init-called
```

Comment:
It looks like `ObjectTracker` is used as a mixin, but it has an `__init__`. Other than suppressing the pylint warning, I'm not sure there's anything else to do. We could rename the `__init__` to `_fake_init`, but then we would have to suppress `attribute-defined-outside-init`. I would leave it as it is unless there's something I'm missing.

Reply:
I would expect a mixin class to be defined as a parent: `class TCPConnection(BMProto, TLSDispatcher, ObjectTracker):` and for the `__init__`s to be called in an appropriate order, each seeing `*args` and `**kwargs` and each updating self as they go, passing off initialisation to the next parent class using `super(ThisClass, self).__init__(*args, **kwargs)`.

However, if this is working as intended, perhaps re-initialisation is both happening and required? Either way, fixing this bad code smell will require some consideration, refactoring and testing.

Reply:
`super()` wont work without refactoring, I tried.